languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -10069,6 +10069,46 @@ though, and has since been slightly modified)-->
     </rulegroup>
   </category>
   <category id="CAT_GRAMMAIRE" name="Grammaire">
+      <rulegroup id="EXPRESSIONS_FIGEES_TOUT" name="expressions figées tout" default="temp_off">
+          <rule>
+              <pattern>
+                  <token>une</token>
+                  <token>bonne</token>
+                  <token>fois</token>
+                  <token>pour</token>
+                  <marker><token>toute</token></marker>
+              </pattern>
+              <message>Dans cette expression \5 s'accorde implicitement avec toutes les fois. </message>
+              <suggestion>toutes</suggestion>
+              <example correction="toutes">Il faudra que tu règles ce problème une bonne fois pour <marker>toute</marker>.</example>
+          </rule>
+          <rule>
+              <pattern>
+                  <marker>
+                      <token>de</token>
+                      <token>toutes</token>
+                      <token>les</token>
+                      <token>façons</token>
+                  </marker>
+              </pattern>
+              <message>Cette expression est familière est incorrecte.</message>
+              <suggestion>de toute façon</suggestion>
+              <example correction="de toute façon">Il faudra que tu règles ce problème <marker>de toutes les façons</marker>.</example>
+          </rule>
+          <rule>
+              <pattern>
+                  <marker>
+                      <token>toute</token>
+                  </marker>
+                  <token>à</token>
+                  <token regexp="yes">fait|coup</token>
```

Comment:
We'd need antipatterns for cases like "à coup sûr", imagine a sentence like this:

>Tu vois cette tarte-là ? Il la mangera toute à coup sûr.

Reply:
It's a good point! I put an antipattern for this case so there will be no error if the user forget the comma between "toute" et "à coup sûr".